src/openttd.cpp

```diff
@@ -603,6 +614,42 @@ int openttd_main(int argc, char *argv[])
 		case 'P':
 			scanner->join_company_password = mgo.opt;
 			break;
+		case 'U': {
+			char* p = mgo.opt;
+
+			const char* protocol = "openttd://";
+			if (strncmp(p, protocol, strlen(protocol)) != 0) {
+				fprintf(stderr, "Unknown protocol\n");
+				break;
+			}
+
+			p += strlen(protocol);
+
+			char* network_conn = p;
+			p = strchr(p, '?');
+			*p = '\0';
+			p++;
+			scanner->network_conn = network_conn;
+
+			const char* name = nullptr;
+			const char* value = nullptr;
+			while (ParseQueryString(&name, &value, &p)) {
```

Comment:
You need to handle [escaped characters](https://en.wikipedia.org/wiki/Percent-encoding) in the query string. You wouldn't be able to handle passwords with ampersands in them, at the very least.